liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,500 @@
+            .dividedBy(data.tokens[currentToken].cTokenDecimalsMult);
+          const tokenQty = data.supply[currentToken][currentAccount]
+            .multipliedBy(data.tokens[currentToken].exchangeRateMult);
+          ts(
+            `User ${currentAccount
+            } supplied ${Math.round(tokenQty)
+            } ${data.tokens[currentToken].symbol
+            } ( ${Math.round(data.supply[currentToken][currentAccount])
+            } cTokens ) `,
+          );
+        }
+        if (snapshot && snapshot[2].toString() !== '0') {
+          data.borrow[currentToken][currentAccount] = BigNumber(snapshot[2].toString())
+            .dividedBy(data.tokens[currentToken].tokenDecimalsMult);
+          ts(
+            `User ${currentAccount
+            } borrowed ${Math.round(data.borrow[currentToken][currentAccount])
+            } ${data.tokens[currentToken].symbol}`,
+          );
+        }
+        /* eslint-enable no-param-reassign */
+        return null;
+      }));
+    }));
+    // #endregion
+
+    // #region Update all token prices via 1inch for now.
+    ts('Updating token prices and collateral factors');
+    // To-Do: Switch the primary feed back to ChainLink
+    await Promise.all(Object.keys(data.tokens).map(async (currentToken) => {
+      const price = await oneInchContract.getRateToEth(data.tokens[currentToken].underlying, 0);
+      // Adjust for native decimals
+      const oneInchMult = BigNumber(10).pow(36 - data.tokens[currentToken].tokenDecimals);
+      /* eslint-disable no-param-reassign */
+      data.tokens[currentToken].price = BigNumber(price.toString())
+        .dividedBy(oneInchMult);
+      // Update the Collateral Factor
+      const market = await comptrollerContract.markets(currentToken);
+      data.tokens[currentToken].collateralMult = BigNumber(market[1].toString())
+        .dividedBy(BigNumber(10).pow(18));
+      /* eslint-enable no-param-reassign */
+      ts(
+        `Updated price of ${data.tokens[currentToken].symbol
+        } is ${data.tokens[currentToken].price.toString()
+        } ETH with a collateral factor of ${data.tokens[currentToken].collateralMult}`,
+      );
+    }));
+    // #endregion
+
+    // #region Calculate health on all accounts
+    ts('Health recalculation');
+    Object.keys(data.accounts).forEach((account) => {
+      let borrowBalance = BigNumber(0);
+      let supplyBalance = BigNumber(0);
+      Object.keys(data.tokens).forEach((token) => {
+        if (data.supply[token][account]) {
+          // Supply balances are stored in cTokens and need to be multiplied by the
+          //   exchange rate and reduced by the collateral factor
+          supplyBalance = supplyBalance.plus(data.supply[token][account]
+            .multipliedBy(data.tokens[token].exchangeRateMult)
+            .multipliedBy(data.tokens[token].price)
+            .multipliedBy(data.tokens[token].collateralMult));
```

Comment:
This comment does not appear to match the code it describes: `reduced by the collateral factor` implies either a subtraction or division, but this uses a multiplication.

Reply:
Corrected in commit 79512957480ad93f994c11aa269380233ad83787